airflow/hooks/dbapi_hook.py

```diff
@@ -27,9 +27,10 @@
 
 from airflow.hooks.base_hook import BaseHook
 from airflow.exceptions import AirflowException
+from airflow.hooks.abstract_db_hook import AbstractDbHook
 
 
-class DbApiHook(BaseHook):
+class DbApiHook(BaseHook, AbstractDbHook):
```

Comment:
Yeah that's indeed strange.

I believe there are several classes which do not extend DBApiHook but still implement those methods

I do not know the full history and why those methods were added to base hook at all. So i only wanted to extract the common part of Not Implemented to be on a safe side rather than move the methods up to the DbApiHook. 

I could not find any "dynamic" use of those methods (that would justify a bit why they are there. But maybe there is a good reason for it that I am not aware of?

Reply:
Not likely. None that I'm aware of either, so lets fix it "right" :)